app/src/main/java/org/mifos/selfserviceapp/ui/fragments/HomeFragment.java

```diff
@@ -0,0 +1,311 @@
+//            @Override
+//            public void onClick(View v) {
+//                ((BaseActivity) getActivity()).
+//                      replaceFragment(LoanRepaymentScheduleFragment.
+//                      newInstance(Constants.CLIENT_ID), false, R.id.container);
+//            }
+//        });
+
+        return rootView;
+    }
+
+    @OnClick(R.id.noAccountIcon)
+    void reloadOnError() {
+        ll_error.setVisibility(View.GONE);
+        homePresenter.loadInfo();
+        homePresenter.loadClient();
+    }
+
+    @Override
+    public void onRefresh() {
+        ll_error.setVisibility(View.GONE);
+        homePresenter.loadInfo();
+        homePresenter.loadClient();
+    }
+
+
+    public void showEmptyAccounts(String emptyAccounts) {
+        ll_error.setVisibility(View.VISIBLE);
+        noAccountText.setText(emptyAccounts);
+//        iv_userPicture.setImageResource(R.drawable.ic_clients);
+    }
+
+    @Override
+    public void showError(String errorMessage) {
+        ll_error.setVisibility(View.VISIBLE);
+        noAccountText.setText(errorMessage);
+//        iv_userPicture.setImageResource(R.drawable.ic_clients);
+        Toast.makeText(getActivity(), errorMessage, Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void showProgress() {
+
+    }
+
+    @Override
+    public void showClientInfo(Client client) {
+        tv_client_name.setText("Hello, " + client.getFirstname());
+        iv_profile_pic.setImageResource(R.drawable.ic_clients);
+    }
+
+    @Override
+    public void showInfo(List<LoanAccount> loanAccounts,
+                         List<SavingAccount> savingAccounts,
+                         List<ShareAccount> shareAccounts) {
+
+        this.savingAccounts = savingAccounts;
+        this.loanAccounts = loanAccounts;
+        this.shareAccounts = shareAccounts;
+
+        showSavingTotal(savingAccounts);
+        showShareTotal(shareAccounts);
+        showLoanTotal(loanAccounts);
+
+    }
+
+
+    @Override
+    public void showSavingTotal(List<SavingAccount> savingAccounts) {
+        double total = 0;
+        for (SavingAccount s: savingAccounts) {
+            total += s.getAccountBalance();
+        }
+        DecimalFormat df = new DecimalFormat("#.##");
+        tv_savings.setText("Total: " + Double.parseDouble(df.format(total)));
+    }
+
+    @Override
+    public void showShareTotal(List<ShareAccount> shareAccounts) {
+        double total = 0;
+        for (ShareAccount sh: shareAccounts) {
+            total += sh.getTotalApprovedShares();
+        }
+        DecimalFormat df = new DecimalFormat("#.##");
+        tv_shares.setText("Total: " + Double.parseDouble(df.format(total)));
+    }
+
+    @Override
+    public void showLoanTotal(final List<LoanAccount> loanAccounts) {
+        for (final LoanAccount l : loanAccounts) {
+            Thread t = new Thread () {
+                public void run() {
+                    homePresenter.loadLoanAccountDetails(l.getId());
```

Comment:
its not because you don't have access to the total outstanding because you need to get the summary first. I asked this in on the issue. Its why you need " this.totalLoans += loanAccount.getSummary().getTotalOutstanding();" later